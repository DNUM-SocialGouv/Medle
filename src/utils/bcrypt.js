import { compare, hash } from "bcryptjs"

// FaÃ§ade for bcrypt functions
export const hashPassword = async (password) => await hash(password, 10)

export const compareWithHash = async (password, hash) => compare(password, hash)
