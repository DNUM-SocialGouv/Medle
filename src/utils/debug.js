import { useRouter } from "next/router"
import { useEffect, useRef } from "react"

import { logInfo } from "./logger"

// useEffect(() => {
//    logInfo("XX dans useEffect")
//    const routeChangeComplete = async () => {
//       logInfo("XXX dans routeChangeComplete")
//       window.scrollTo(0, 0)

//       logInfo("mainRef", mainRef)

//       if (mainRef && mainRef.current) {
//          mainRef.current.focus()
//       }
//    }
//    router.events.on("routeChangeComplete", routeChangeComplete)

//    return () => {
//       router.events.off("routeChangeComplete", routeChangeComplete)
//    }
// }, [router.events])

export const DebugRouter = () => {
  const router = useRouter()

  router.events.on("routeChangeStart", (url) => {
    logInfo("event routeChangeStart", url)
  })
  router.events.on("routeChangeComplete", (url) => {
    logInfo("event routeChangeComplete", url)
  })
  router.events.on("routeChangeError", (err, url) => {
    logInfo("event routeChangeError", url, err)
  })

  return null
}
// Usage
// function MyComponent(props) {
//    useTraceUpdate(props)
//    return <div>{props.children}</div>
// }
export const useTraceUpdate = (props) => {
  const prev = useRef(props)
  useEffect(() => {
    const changedProps = Object.entries(props).reduce((ps, [k, v]) => {
      if (prev.current[k] !== v) {
        ps[k] = [prev.current[k], v]
      }
      return ps
    }, {})
    if (Object.keys(changedProps).length > 0) {
      logInfo("Changed props:", changedProps)
    }
    prev.current = props
  })
}

// usage: <Logger label="Mon composant Ã  loguer" />
export const Logger = (props) => {
  logInfo(`${props.label} rendered`)
  return null // irrelevant
}
