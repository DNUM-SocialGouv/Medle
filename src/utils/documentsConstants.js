export const footerDocument = "footer-document-"

export const footerDocumentAccessibilite = footerDocument + "accessibilite"
export const footerDocumentMentionsLegales = footerDocument + "mentions-legales"
export const footerDocumentDonneesPersonnelles = footerDocument + "donnees-personnelles"
export const footerDocumentGestionCookies = footerDocument + "gestion-cookies"
export const footerDocumentFAQ = footerDocument + "faq"
