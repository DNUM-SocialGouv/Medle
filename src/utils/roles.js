export const HOSPITAL_DETAILS_MANAGEMENT = "HOSPITAL_DETAILS_MANAGEMENT" // peut modifier les informations de l'ETS
export const ACT_CONSULTATION = "ACT_CONSULTATION" // peut consulter les actes d'un ou plusieurs ETS
export const ACT_MANAGEMENT = "ACT_MANAGEMENT" // peut ajouter/modifier les actes pour un ou plusieurs ETS
export const EMPLOYMENT_CONSULTATION = "EMPLOYMENT_CONSULTATION" // peut consulter les ETP pour un ou plusieurs ETS
export const EMPLOYMENT_MANAGEMENT = "EMPLOYMENT_MANAGEMENT" // peut ajouter/modifier les ETP pour un ou plusieurs ETS
export const STATS_LOCAL = "STATS_LOCAL" // peut voir les stats pour un ou plusieurs ETS
export const STATS_GLOBAL = "STATS_GLOBAL" // peut voir les stats publiques
export const ADD_USER_OPERATOR_ACT = "ADD_USER_OPERATOR_ACT" // peut ajouter un utilisateur gestionnaire d'actes pour un ou plusieurs ETS
export const ADD_USER_OPERATOR_EMPLOYMENT = "ADD_USER_OPERATOR_EMPLOYMENT" // peut ajouter un utilisateur gestionnaire d'ETP pour un ou plusieurs ETS
export const ADD_USER_ADMIN_HOSPITAL = "ADD_USER_ADMIN_HOSPITAL" // peut ajouter un utilisateur administrateur d'un ETS pouvant créer des administrateurs d'actes et des administrateurs d'ETP pour un ou plusieurs ETS
export const ADD_TYPE_VIOLENCE = "ADD_TYPE_VIOLENCE" // peut ajouter des types de violences occasionnellement (attentat, etc.)
export const NO_PRIVILEGE_REQUIRED = "NO_PRIVILEGE_REQUIRED"
export const ADMIN = "ADMIN"

export const ADMIN_HOSPITAL = "ADMIN_HOSPITAL"
export const SUPER_ADMIN = "SUPER_ADMIN"

export const PRIVILEGES = [
   HOSPITAL_DETAILS_MANAGEMENT,
   ACT_CONSULTATION,
   ACT_MANAGEMENT,
   EMPLOYMENT_CONSULTATION,
   EMPLOYMENT_MANAGEMENT,
   STATS_LOCAL,
   STATS_GLOBAL,
   ADD_USER_OPERATOR_ACT,
   ADD_USER_OPERATOR_EMPLOYMENT,
   ADD_USER_ADMIN_HOSPITAL,
   ADD_TYPE_VIOLENCE,
   NO_PRIVILEGE_REQUIRED,
   ADMIN,
]

export const ROLES = {
   ADMIN_HOSPITAL: [
      // ex: le responsable d'un UMJ et/ou d'un IML
      ADD_USER_OPERATOR_ACT,
      ADD_USER_OPERATOR_EMPLOYMENT,
      HOSPITAL_DETAILS_MANAGEMENT,
      ACT_CONSULTATION,
      ACT_MANAGEMENT,
      EMPLOYMENT_CONSULTATION,
      EMPLOYMENT_MANAGEMENT,
      STATS_LOCAL,
      STATS_GLOBAL,
      ADMIN,
   ],
   OPERATOR_ACT: [ACT_CONSULTATION, ACT_MANAGEMENT, EMPLOYMENT_CONSULTATION, STATS_LOCAL, STATS_GLOBAL], // un gestionnaire d'acte
   OPERATOR_EMPLOYMENT: [EMPLOYMENT_CONSULTATION, EMPLOYMENT_MANAGEMENT, ACT_CONSULTATION, STATS_LOCAL, STATS_GLOBAL], // un gestionnaire d'ETP
   OPERATOR_GENERIC: [
      EMPLOYMENT_CONSULTATION,
      EMPLOYMENT_MANAGEMENT,
      ACT_CONSULTATION,
      ACT_MANAGEMENT,
      STATS_LOCAL,
      STATS_GLOBAL,
   ], // un gestionnaire d'actes et d'ETP
   GUEST_HOSPITAL: [ACT_CONSULTATION, EMPLOYMENT_CONSULTATION, STATS_LOCAL, STATS_GLOBAL],
   SUPER_ADMIN: PRIVILEGES, // le super admin qui a tous les privilèges
   REGIONAL_SUPERVISOR: [ACT_CONSULTATION, EMPLOYMENT_CONSULTATION, STATS_LOCAL, STATS_GLOBAL], // un ARS par exemple
   PUBLIC_SUPERVISOR: [STATS_LOCAL, STATS_GLOBAL], // ex: un superviseur public, tel que ARS, Ministère de la justice
}

export const ROLES_DESCRIPTION = {
   ADMIN_HOSPITAL: "Administrateur d'UMJ ou d'IML", // no scope, 1 hospital_id
   OPERATOR_ACT: "Gestionnaire d'actes", // no scope, 1 hospital_id
   OPERATOR_EMPLOYMENT: "Gestionnaire d'emplois", // no scope, 1 hospital_id
   OPERATOR_GENERIC: "Gestionnaire d'actes et ETP", // no scope, 1 hospital_id
   GUEST_HOSPITAL: "Invité", // no scope, 1 hospital_id
   SUPER_ADMIN: "Administrateur de Medlé", // no scope, no hospital_id
   PUBLIC_SUPERVISOR: "Superviseur publique", // no scope, no hospital_id
   REGIONAL_SUPERVISOR: "Superviseur de plusieurs UMJ ou IML", // n hospitals in scope, no hospital_id
}

export const START_PAGES = {
   OPERATOR_EMPLOYMENT: "/fillEmployments",
   OPERATOR_ACT: "/actsList",
   OPERATOR_GENERIC: "/actsList",
}

export const isAllowed = (role, privilege) =>
   privilege === NO_PRIVILEGE_REQUIRED || (ROLES[role] && ROLES[role].includes(privilege))
