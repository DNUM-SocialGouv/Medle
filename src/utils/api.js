import {
   STATUS_200_OK,
   STATUS_401_UNAUTHORIZED,
   STATUS_403_FORBIDDEN,
   STATUS_405_METHOD_NOT_ALLOWED,
   STATUS_500_INTERNAL_SERVER_ERROR,
} from "./http"
import { isAllowed, NO_PRIVILEGE_REQUIRED } from "./roles"
import { checkToken, decodeToken } from "./jwt"
import { APIError } from "./errors"

// see Micro.sendError
export const sendError = (req, res, error) => {
   const statusCode = error.statusCode || error.status
   const message = statusCode ? error.message : "Internal Server Error"

   res.status(statusCode || STATUS_500_INTERNAL_SERVER_ERROR).json({ message })

   if (error instanceof Error) {
      console.error(error.stack)
   } else {
      console.warn("thrown error must be an instance Error")
   }
}

export const sendSuccess = (req, res) => res.status(STATUS_200_OK)

// see Micro.createError
export const createError = (code, message, original) => {
   const error = new Error(message)

   error.statusCode = code
   error.originalError = original

   return error
}

export const sendAPIError = (error, res) => {
   if (error instanceof APIError) {
      return res.status(error.status).json(error)
   }
   // fallback error
   return res
      .status(STATUS_500_INTERNAL_SERVER_ERROR)
      .json({ message: `Erreur de base de donnÃ©e / ${error}`, status: STATUS_500_INTERNAL_SERVER_ERROR })
}

export const checkValidUserWithPrivilege = (privilege, req) => {
   const { token } = req.cookies

   try {
      if (!token) {
         throw new APIError({
            message: "Non authentified user",
            status: STATUS_401_UNAUTHORIZED,
         })
      }

      const currentUser = checkToken(token)

      if (privilege !== NO_PRIVILEGE_REQUIRED && !isAllowed(currentUser.role, privilege)) {
         throw new APIError({
            message: `Not allowed role  (${currentUser.email ? currentUser.email : "unknown user"})`,
            status: STATUS_403_FORBIDDEN,
         })
      } else {
         return currentUser
      }
   } catch (error) {
      let email
      try {
         // Let's try to get user informations even if the token is not valid
         const currentUser = decodeToken(token)
         email = currentUser.email
      } catch (error) {
         console.error("Token couldn't been decoded")
      }
      throw new APIError({
         message: `Invalid token for user (${email ? email : "unknown user"})`,
         status: STATUS_401_UNAUTHORIZED,
      })
   }
}

export const sendMethodNotAllowedError = res => {
   return res.status(STATUS_405_METHOD_NOT_ALLOWED).end()
}
