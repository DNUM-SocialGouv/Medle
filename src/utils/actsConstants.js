import joursFeries from "@socialgouv/jours-feries"
import moment from "moment"

import AsylumSeekerProfile from "../components/profiles/AsylumSeekerProfile"
import BoneAgeProfile from "../components/profiles/BoneAgeProfile"
import CriminalCourtProfile from "../components/profiles/CriminalCourtProfile"
import CustodyProfile from "../components/profiles/CustodyProfile"
import DeceasedProfile from "../components/profiles/DeceasedProfile"
import DrunkProfile from "../components/profiles/DrunkProfile"
import ReconstitutionProfile from "../components/profiles/ReconstitutionProfile"
import RestrainedProfile from "../components/profiles/RestrainedProfile"
import RoadRelatedExaminationProfile from "../components/profiles/RoadRelatedExaminationProfile"
import VictimProfile from "../components/profiles/VictimProfile"
import { ISO_DATE } from "./date"

export const orderedProfileValues = [
  "Victime (vivante)",
  "Gardé.e à vue",
  "Personne pour âge osseux (hors GAV)",
  "Examen pour OFPRA",
  "Personne décédée",
  {
    title: "Autre activité",
    subValues: ["Personne retenue", "Examen lié à la route", "IPM", "Assises", "Reconstitution"],
  },
]

export const profiles = {
  "Victime (vivante)": VictimProfile,
  "Personne décédée": DeceasedProfile,
  "Gardé.e à vue": CustodyProfile,
  "Personne pour âge osseux (hors GAV)": BoneAgeProfile,
  "Examen pour OFPRA": AsylumSeekerProfile,
  "Autre activité/Personne retenue": RestrainedProfile,
  "Autre activité/Examen lié à la route": RoadRelatedExaminationProfile,
  "Autre activité/IPM": DrunkProfile,
  "Autre activité/Assises": CriminalCourtProfile,
  "Autre activité/Reconstitution": ReconstitutionProfile,
}

export const livingProfiles = [
  { value: "Victime (vivante)", label: "Victime" },
  { value: "Gardé.e à vue", label: "Gardé.e à vue" },
  { value: "Personne pour âge osseux (hors GAV)", label: "Personne pour âge osseux" },
  { value: "Examen pour OFPRA", label: "Examen pour OFPRA" },
  { value: "Autre activité/Personne retenue", label: "Personne retenue" },
  { value: "Autre activité/Examen lié à la route", label: "Examen lié à la route" },
  { value: "Autre activité/IPM", label: "IPM" },
]

const periodOfDayValues = {
  week: {
    title: "lun.-ven.",
    period: [
      {
        title: "Nuit profonde",
        subTitle: "(00h-8h30)",
      },
      {
        title: "Journée",
        subTitle: "(8h30-18h30)",
      },
      {
        title: "Soirée",
        subTitle: "(18h30-00h)",
      },
    ],
  },
  saturday: {
    title: "sam.",
    period: [
      {
        title: "Nuit profonde",
        subTitle: "(00h-8h30)",
      },
      {
        title: "Matin",
        subTitle: "(8h30-12h30)",
      },
      {
        title: "Après-midi",
        subTitle: "(12h30-18h)",
      },
      {
        title: "Soirée",
        subTitle: "(18h-00h)",
      },
    ],
  },
  sunday: {
    title: "dim. et férié",
    period: [
      {
        title: "Nuit profonde",
        subTitle: "(00h-8h30)",
      },
      {
        title: "Journée",
        subTitle: "(08h30-18h30)",
      },
      {
        title: "Soirée",
        subTitle: "(18h30-00h)",
      },
    ],
  },
}

const getSituationDate = (dateStr) => {
  const publicHolidays = joursFeries(moment(dateStr).format("YYYY"))

  const publicHolidaysDates = Object.values(publicHolidays).map((elt) => moment(elt).format(ISO_DATE))

  if (publicHolidaysDates.includes(dateStr)) {
    return "sunday"
  }

  const dayInt = moment(dateStr).day()
  switch (dayInt) {
    case 0:
      return "sunday"
    case 6:
      return "saturday"
    default:
      return "week"
  }
}

export { getSituationDate, periodOfDayValues }
