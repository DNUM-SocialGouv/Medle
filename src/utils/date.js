import getConfig from "next/config"
import moment from "moment"

import { logError } from "../utils/logger"

const { publicRuntimeConfig } = getConfig() || {}

export const NAME_MONTHS = {
  "01": "janvier",
  "02": "février",
  "03": "mars",
  "04": "avril",
  "05": "mai",
  "06": "juin",
  "07": "juillet",
  "08": "août",
  "09": "septembre",
  "10": "octobre",
  "11": "novembre",
  "12": "décembre",
}

export const FORMAT_DATE = "DD/MM/YYYY"
export const ISO_DATE = "YYYY-MM-DD"
export const ISO_TIME = "YYYY-MM-DDTHH:mm:ssZ"

export const now = () =>
  (publicRuntimeConfig?.TEST_CURRENT_DATE && moment(publicRuntimeConfig?.TEST_CURRENT_DATE, FORMAT_DATE)) || moment()

export const isValidIsoDate = (date) => date && moment(date, ISO_DATE, true).isValid()

export const frToIso = (date) => {
  const parts = date.split("/")

  if (parts.length !== 3) {
    logError("Problème de parsing de date")
    return date
  }
  const [day, month, year] = parts
  return `${year}-${month}-${day}`
}

export const isoToFr = (date) => {
  const parts = date.split("-")

  if (parts.length !== 3) {
    logError("Problème de parsing de date")
    return date
  }
  const [year, month, day] = parts
  return `${day}/${month}/${year}`
}
