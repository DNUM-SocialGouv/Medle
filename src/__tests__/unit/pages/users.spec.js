import React from "react"
import renderer from "react-test-renderer"
import * as nextRouter from "next/router"

import UserDetail from "../../../pages/administration/users/[id]"
import { SUPER_ADMIN } from "../../../utils/roles"

// API dependency for pages's getInitialProps

it("should renders UserDetail unchanged", () => {
  /* eslint-disable  no-import-assign */
  nextRouter.useRouter = jest.fn()
  nextRouter.useRouter.mockImplementation(() => ({ query: { id: "3" } }))

  const currentUser = {
    id: 3,
    role: SUPER_ADMIN,
    hospital: null,
  }
  const tree = renderer.create(<UserDetail initialUser={{}} currentUser={currentUser} />).toJSON()
  expect(tree).toMatchSnapshot()
})
