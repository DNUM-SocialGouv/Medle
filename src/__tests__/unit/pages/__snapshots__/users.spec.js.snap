// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders UserDetail unchanged 1`] = `
<div
  className="jsx-1821637326 d-flex flex-column justifiy-content-between min-vh-100"
>
  <header
    className="border-bottom"
  >
    <nav
      className="navbar navbar-expand-md navbar-light"
    >
      <a
        className="navbar-brand"
      >
        <img
          alt="Logo"
          src="/images/logo.png"
          title="Logo"
          width="100"
        />
      </a>
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="pt-2 mt-2 ml-auto d-flex justify-content-end align-items-md-center align-items-start pt-md-0 navbar-nav"
        >
          <li
            className="dropdown nav-item"
            onKeyDown={[Function]}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root text-black-50"
                focusable="false"
                viewBox="0 0 24 24"
                width={30}
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
              <span
                className="d-sm-inline d-md-none text-black-50"
              >
                 Mon compte
              </span>
            </a>
            <div
              aria-hidden={true}
              className="dropdown-menu dropdown-menu-right"
              role="menu"
              tabIndex="-1"
            >
              <button
                className="dropdown-item"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                undefined undefined
                 
              </button>
              <div
                className="dropdown-divider"
                onClick={[Function]}
                tabIndex="-1"
              />
              <a
                href="/profile"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Profil
                </button>
              </a>
              <div
                className="dropdown-divider"
                onClick={[Function]}
                tabIndex="-1"
              />
              <button
                className="dropdown-item"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                Se déconnecter
              </button>
            </div>
          </li>
        </ul>
      </div>
    </nav>
  </header>
  <div
    className="jsx-1821637326 d-flex"
    id="wrapper"
  >
    <div
      className="jsx-1821637326 border-right"
      id="sidebar-wrapper"
    >
      <div
        className="jsx-1576199254 list-group list-group-flush text-center"
      >
        <a
          className="jsx-1576199254 list-group-item list-group-item-action selected"
          href="/administration/users"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root text-black-50"
            focusable="false"
            viewBox="0 0 24 24"
            width={30}
          >
            <path
              d="M9 11.75c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zm6 0c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8 0-.29.02-.58.05-.86 2.36-1.05 4.23-2.98 5.21-5.37C11.07 8.33 14.05 10 17.42 10c.78 0 1.53-.09 2.25-.26.21.71.33 1.47.33 2.26 0 4.41-3.59 8-8 8z"
            />
          </svg>
          <br
            className="jsx-1576199254"
          />
          Utilisateurs
        </a>
        <a
          className="jsx-1576199254 list-group-item list-group-item-action unselected"
          href="/administration/hospitals"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
            width={30}
          >
            <path
              d="M17 11V3H7v4H3v14h8v-4h2v4h8V11h-4zM7 19H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm4 4H9v-2h2v2zm0-4H9V9h2v2zm0-4H9V5h2v2zm4 8h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm4 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
            />
          </svg>
           
          <br
            className="jsx-1576199254"
          />
          Établissements
        </a>
        <a
          className="jsx-1576199254 list-group-item list-group-item-action unselected"
          href="/administration/askers"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
            width={30}
          >
            <path
              d="M4 10h3v7H4zM10.5 10h3v7h-3zM2 19h20v3H2zM17 10h3v7h-3zM12 1L2 6v2h20V6z"
            />
          </svg>
           
          <br
            className="jsx-1576199254"
          />
          Demandeurs
        </a>
        <a
          className="jsx-1576199254 list-group-item list-group-item-action unselected"
          href="/administration/attacks"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
            width={30}
          >
            <path
              d="M13.5.67s.74 2.65.74 4.8c0 2.06-1.35 3.73-3.41 3.73-2.07 0-3.63-1.67-3.63-3.73l.03-.36C5.21 7.51 4 10.62 4 14c0 4.42 3.58 8 8 8s8-3.58 8-8C20 8.61 17.41 3.8 13.5.67zM11.71 19c-1.78 0-3.22-1.4-3.22-3.14 0-1.62 1.05-2.76 2.81-3.12 1.77-.36 3.6-1.21 4.62-2.58.39 1.29.59 2.65.59 4.04 0 2.65-2.15 4.8-4.8 4.8z"
            />
          </svg>
           
          <br
            className="jsx-1576199254"
          />
          Attentats
        </a>
        <a
          className="jsx-1576199254 list-group-item list-group-item-action unselected"
          href="/administration/acts"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
            width={30}
          >
            <path
              d="M18 17H6v-2h12v2zm0-4H6v-2h12v2zm0-4H6V7h12v2zM3 22l1.5-1.5L6 22l1.5-1.5L9 22l1.5-1.5L12 22l1.5-1.5L15 22l1.5-1.5L18 22l1.5-1.5L21 22V2l-1.5 1.5L18 2l-1.5 1.5L15 2l-1.5 1.5L12 2l-1.5 1.5L9 2 7.5 3.5 6 2 4.5 3.5 3 2v20z"
            />
          </svg>
           
          <br
            className="jsx-1576199254"
          />
          Actes
        </a>
        <a
          className="jsx-1576199254 list-group-item list-group-item-action unselected"
          href="/administration/messages"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
            width={30}
          >
            <path
              d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
            />
          </svg>
           
          <br
            className="jsx-1576199254"
          />
          Messages
        </a>
        <a
          className="jsx-1576199254 list-group-item list-group-item-action"
          href="/statistics"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
            width={30}
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
           
          <br
            className="jsx-1576199254"
          />
          Retour
        </a>
      </div>
    </div>
    <div
      className="jsx-1821637326"
      id="page-content-wrapper"
    >
      <main
        className="jsx-1821637326 pb-5"
      >
        <div
          className="mt-5 mb-4 container"
          style={
            Object {
              "maxWidth": 720,
            }
          }
        >
          <div
            className="d-flex justify-content-between"
          >
            <a
              href="/administration/users"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                style={
                  Object {
                    "width": 15,
                  }
                }
                viewBox="0 0 24 24"
                width={30}
              >
                <path
                  d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
                />
              </svg>
              Retour
            </a>
            <h1
              className="StyledComponents__Title1-yzg90z-6 kaiHmj"
            >
              Utilisateur
            </h1>
            <span>
               
            </span>
          </div>
          
          <form
            className="mt-4"
            onSubmit={[Function]}
          >
            <div
              className="row form-group"
            >
              <label
                className="col-sm-3 col-form-label"
                htmlFor="id"
              >
                Id
              </label>
              <div
                className="col-sm-9"
              >
                <input
                  className="form-control"
                  disabled={true}
                  id="id"
                  name="id"
                  type="text"
                />
              </div>
            </div>
            <div
              className="row form-group"
            >
              <label
                className="col-sm-3 col-form-label"
                htmlFor="firstName"
              >
                Prénom
              </label>
              <div
                className="col-sm-9"
              >
                <input
                  className="form-control"
                  id="firstName"
                  name="firstName"
                  type="text"
                />
              </div>
            </div>
            <div
              className="row form-group"
            >
              <label
                className="col-sm-3 col-form-label"
                htmlFor="lastName"
              >
                Nom 
                <span
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  *
                </span>
              </label>
              <div
                className="col-sm-9"
              >
                <input
                  aria-invalid={false}
                  className="form-control"
                  id="lastName"
                  name="lastName"
                  type="text"
                />
                <div
                  className="invalid-feedback"
                />
              </div>
            </div>
            <div
              className="row form-group"
            >
              <label
                className="col-sm-3 col-form-label"
                htmlFor="email"
              >
                Courriel 
                <span
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  *
                </span>
              </label>
              <div
                className="col-sm-9"
              >
                <input
                  aria-invalid={false}
                  className="form-control"
                  id="email"
                  name="email"
                  type="text"
                />
                <div
                  className="invalid-feedback"
                />
              </div>
            </div>
            <div
              className="row form-group"
            >
              <label
                className="col-sm-3 col-form-label"
                htmlFor="role"
              >
                Rôle 
                <span
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  *
                </span>
              </label>
              <div
                className="col-sm-9"
              >
                <div
                  className=" css-2b097c-container"
                  onKeyDown={[Function]}
                >
                  <div
                    className=" css-yk16xz-control"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <div
                      className=" css-g1d714-ValueContainer"
                    >
                      <div
                        className=" css-1wa3eu0-placeholder"
                      >
                        Choisissez un rôle
                      </div>
                      <div
                        className="css-b8ldur-Input"
                      >
                        <div
                          className=""
                          style={
                            Object {
                              "display": "inline-block",
                            }
                          }
                        >
                          <input
                            aria-autocomplete="list"
                            autoCapitalize="none"
                            autoComplete="off"
                            autoCorrect="off"
                            disabled={false}
                            id="react-select-2-input"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            spellCheck="false"
                            style={
                              Object {
                                "background": 0,
                                "border": 0,
                                "boxSizing": "content-box",
                                "color": "inherit",
                                "fontSize": "inherit",
                                "label": "input",
                                "opacity": 1,
                                "outline": 0,
                                "padding": 0,
                                "width": "1px",
                              }
                            }
                            tabIndex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style={
                              Object {
                                "height": 0,
                                "left": 0,
                                "overflow": "scroll",
                                "position": "absolute",
                                "top": 0,
                                "visibility": "hidden",
                                "whiteSpace": "pre",
                              }
                            }
                          >
                            
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      className=" css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        className=" css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        className=" css-tlfecz-indicatorContainer"
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                      >
                        <svg
                          aria-hidden="true"
                          className="css-6q0nyr-Svg"
                          focusable="false"
                          height={20}
                          viewBox="0 0 20 20"
                          width={20}
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="justify-content-center d-flex"
            >
              <button
                aria-label={null}
                className="px-4 mt-3 mr-3 btn btn-outline-primary"
                onClick={[Function]}
                onMouseEnter={[Function]}
                type="button"
              >
                Annuler
              </button>
              <button
                aria-label={null}
                className="px-4 mt-3 btn btn-primary"
                onClick={[Function]}
              >
                Ajouter
              </button>
            </div>
          </form>
          <div />
        </div>
      </main>
    </div>
  </div>
  <footer
    className="jsx-1982338225 pt-4 pb-5 m-0"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="mr-5 col"
        >
          <h4
            className="jsx-1982338225"
          >
            Medle.fabrique.social.gouv.fr
          </h4>
          Un service proposé par la
           
          <a
            className="jsx-1982338225"
            href="https://solidarites-sante.gouv.fr/ministere/organisation/organisation-des-directions-et-services/article/organisation-de-la-direction-generale-de-l-offre-de-soins-dgos"
            rel="noreferrer noopener"
            target="_blank"
          >
            DGOS
          </a>
           
          (ministère de la santé) et
           
          <a
            className="jsx-1982338225"
            href="https://www.fabrique.social.gouv.fr/"
            rel="noopener noreferrer"
            target="_blank"
          >
            la fabrique numérique des ministères sociaux
          </a>
           
          (
          <a
            className="jsx-1982338225"
            href="https://beta.gouv.fr/"
            rel="noopener noreferrer"
            target="_blank"
          >
            beta.gouv.fr
          </a>
          )
        </div>
        <div
          className="mt-4 mt-md-0 col"
        >
          <ul
            className="jsx-1982338225 pl-0 list-unstyled"
          >
            <li
              className="jsx-1982338225"
            >
              <a
                className="jsx-1982338225"
                href="/statistics"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Statistiques
              </a>
            </li>
            <li
              className="jsx-1982338225"
            >
              <a
                className="jsx-1982338225"
                href="mailto:contact.medle@fabrique.social.gouv.fr"
              >
                Contactez‑nous
              </a>
            </li>
            <li
              className="jsx-1982338225"
            >
              <a
                className="jsx-1982338225"
                href="/faq"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                FAQ
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
